packages/panzer/adapters-stk/test/response_library/tpetra_response_library2.cpp

```diff
@@ -0,0 +1,527 @@
+
+    tMap.set<short>("not char");
+    TEST_EQUALITY(tMap.get<short>(),"not char");
+  }
+
+  struct RespFactoryFunc_Builder {
+    MPI_Comm comm;
+    Teuchos::RCP<panzer::LinearObjFactory<panzer::Traits> > linearObjFactory;
+    Teuchos::RCP<const panzer::GlobalIndexer> globalIndexer;
+
+    template <typename T>
+    Teuchos::RCP<ResponseEvaluatorFactoryBase> build() const
+    { return Teuchos::rcp(new ResponseEvaluatorFactory_Functional<T,int,int>(comm,1,true,"",linearObjFactory)); }
+  };
+
+  TEUCHOS_UNIT_TEST(response_library2, test)
+  {
+    std::vector<Teuchos::RCP<panzer::PhysicsBlock> > physics_blocks;
+    panzer::ClosureModelFactory_TemplateManager<panzer::Traits> cm_factory;
+    Teuchos::ParameterList closure_models("Closure Models");
+    Teuchos::ParameterList user_data("User Data");
+
+    // setup and evaluate ResponseLibrary
+    ///////////////////////////////////////////////////
+
+    out << "Adding responses" << std::endl;
+
+    std::pair< RCP<ResponseLibrary<Traits> >, RCP<panzer::LinearObjFactory<panzer::Traits> > > data
+          = buildResponseLibrary(physics_blocks,cm_factory,closure_models,user_data);
+    RCP<ResponseLibrary<Traits> > rLibrary = data.first;
+    RCP<panzer::LinearObjFactory<panzer::Traits> > lof = data.second;
+
+    RespFactoryFunc_Builder builder;
+    builder.comm = MPI_COMM_WORLD;
+    std::vector<std::string> blocks(1);
+    blocks[0] = "eblock-0_0";
+    rLibrary->addResponse("FIELD_A",blocks,builder);
+    blocks[0] = "eblock-1_0";
+    rLibrary->addResponse("FIELD_B",blocks,builder);
+
+    Teuchos::RCP<ResponseBase> tResp = rLibrary->getResponse<panzer::Traits::Residual>("FIELD_A");
+    Teuchos::RCP<ResponseBase> iResp = rLibrary->getResponse<panzer::Traits::Residual>("FIELD_B");
+
+    TEST_ASSERT(tResp!=Teuchos::null);
+    TEST_ASSERT(iResp!=Teuchos::null);
+
+    TEST_EQUALITY(tResp->getName(),"FIELD_A");
+    TEST_EQUALITY(iResp->getName(),"FIELD_B");
+
+    TEST_EQUALITY(tResp->getLookupName(),"RESPONSE_FIELD_A");
+    TEST_EQUALITY(iResp->getLookupName(),"RESPONSE_FIELD_B");
+
+    TEST_NOTHROW(Teuchos::rcp_dynamic_cast<Response_Functional<panzer::Traits::Residual> >(tResp,true));
+    TEST_NOTHROW(Teuchos::rcp_dynamic_cast<Response_Functional<panzer::Traits::Residual> >(iResp,true));
+
+    RCP<Thyra::VectorBase<double> > tVec;
+    {
+      RCP<const Thyra::VectorSpaceBase<double> > vs = Teuchos::rcp_dynamic_cast<Response_Functional<panzer::Traits::Residual> >(tResp)->getVectorSpace();
+
+      tVec = Thyra::createMember<double>(vs);
+
+      Teuchos::rcp_dynamic_cast<Response_Functional<panzer::Traits::Residual> >(iResp)->setVector(tVec);
+      Teuchos::rcp_dynamic_cast<Response_Functional<panzer::Traits::Residual> >(tResp)->setVector(tVec);
+
+      // test thyra only logic
+      // Disabled until further guidance
+      // TEST_THROW(Teuchos::rcp_dynamic_cast<FakeEpetraEnabledResponse_Functional<panzer::Traits::Residual> >(iResp)->setVector(tVec), std::logic_error);
```

Comment:
@rppawlo this line gives a segfault. I'm not sure I understand what this is meant to do. See line 116 for context about vector type.